syntax = "proto3";
package monitor.proto;

// 导入其他proto文件
import "google/protobuf/empty.proto";  // Google的空消息类型
import "net_info.proto";
import "mem_info.proto";
import "cpu_stat.proto";
import "cpu_softirq.proto";
import "cpu_load.proto";

message MonitorInfo {
    string name = 1;                       // 主机名
    repeated SoftIrq soft_irq = 4;         // 软中断列表（每个CPU一个）
    CpuLoad cpu_load = 5;                  // CPU负载信息
    repeated CpuStat cpu_stat = 6;         // CPU状态列表（每个CPU一个）
    MemInfo mem_info = 7;                  // 内存信息
    repeated NetInfo net_info = 8;         // 网络接口列表
}

// gRPC服务定义
service GrpcManager {
    // 设置监控信息（客户端→服务器）
    rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty) {}
    
    // 获取监控信息（服务器→客户端）
    rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo) {}
}